readme.md

```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+        lastName: 'Doe',
+        message: 'Hi! Could you tell me more about...',
+    });
+
+    // TypeScript error: missing property 'message'
+    submitContactForm({
+        email: 'ex@mple.com',
+        firstName: 'John',
+        lastName: 'Doe',
+    });
+    ```
+    </details>
+
 - [`Readonly<T>`](https://github.com/Microsoft/TypeScript/blob/2961bc3fc0ea1117d4e53bc8e97fa76119bc33e3/src/lib/es5.d.ts#L1415-L1420) - Make all properties in `T` readonly.
+    <details>
+    <summary>
+        Example
+        <a href="https://typescript-play.js.org/?target=6#code/KYOwrgtgBAMg9gcxsAbsANlA3gKClAeQDMiAaPKAEWACMwFz8BRAJxbhcagDEBDAF17ocAXxw4AliH7AWRXgGNgsRAlkBhOCCISE2CiF4RgALigBnfiykIA3BXSoMZ+Eifp7YnAvS9z5lQQ1Fn18BS0dBDMAJWBeABMtdABPAB5XYM1tXQA+ewpwkEsWMAV+DgAKLChDY1IoRzRMERdVDQjdAEpQ-Ch+AAsJcwA6QsioAF5sGqNgesaMKBF7XsIaACtgMuGiFmBgAC9gCoGh0Y6ETpWl0XFCyygx3Vag9uy9Kera0ygAIgBZZIAZTAAAdZABBUGg37zdwvZBNYbUOh6ZZ3LQPdBtEJTEDAADugWCFSel3yAHoKVAACrJcFAhTWUH8KCsdgsMwKXggEBwVl+cy6EB9OBQPYJJLJYY4bGvFjnd7DBaYKauREYYbsjiU6kAdWUvD2UF4NEcoqgwHiElZZKgKCNElN5r8UAJylBjj8wBlZOV7kmgQ16C1bB1OCAA">
+            Playground
+        </a>
+    </summary>
+
+    ```typescript
+    enum LogLevel {
+        Off,
+        Debug,
+        Error,
+        Fatal,
+    };
+
+    interface LoggerConfig {
+        name: string;
+        level: LogLevel;
+    }
+
+    class Logger {
+        config: Readonly<LoggerConfig>;
+
+        constructor({ name, level }: LoggerConfig) {
+            this.config = { name, level };
+            Object.freeze(this.config);
+        }
+    }
+
+    const config: LoggerConfig = { name: 'MySuperApp', level: LogLevel.Debug };
+
+    const logger = new Logger(config);
+
+    // TypeScript Error: cannot assign to readonly.
+    logger.config.level = LogLevel.Error;
+
+    // We are able to edit config variable as we please.
+    config.level = LogLevel.Error;
+    ```
+    </details>
+
 - [`Pick<T, K>`](https://github.com/Microsoft/TypeScript/blob/2961bc3fc0ea1117d4e53bc8e97fa76119bc33e3/src/lib/es5.d.ts#L1422-L1427) - From `T`, pick a set of properties whose keys are in the union `K`.
+    <details>
+    <summary>
+        Example
+        <a href="https://typescript-play.js.org/?target=6#code/JYOwLgpgTgZghgYwgAgIJTMBAbFBvAKGWUzFwC5kBnMKUAcwG4iSALAVwFsAjEOYbJRp0QTFggD24COCG0GzAL4ECYAJ4AHFOkw4IABSgQAbsAgB3ZAF5k+rAGsAPDqy4ANMgBEpXJ+QAfLzAOHj4BTwA+ZgIAehjkACUZABNoZDhkbGAaZAkYdIxXCCpkdioGXJBsNRJgMhQ4EGTkVKoEOg1MKQA6Ahh2EAQukGQjJugXPUMTM3MqAAoNI1MLKkpJ3GmV8wBtAF0ASkoACQAVAFkAGQBRXE4ZMGRCYkkQHLhCvRKbZIkELge3XaEDgkFuEHu4HmnmSwGMngOzGIMAkUGQ81eOSWMwsuXy2O2VAOTxYxDi6Q0WiayAJsxIEjYDU+uCoLGUxCMYHYUBGH10LKUKkxjz5RW+oxSE2ZBmWswWOxYeFq9UonlOmggAGV2sBOiR2GBUcA4NgAISeDzBLi8fiCLwxOBUKgQMBUGKu7oAKw09D8ykOzCAA">
+            Playground
+        </a>
+    </summary>
+
+    ```typescript
+    interface Article {
+        title: string;
+        thumbnail: string;
+        content: string;
+    }
+
+    type ArticlePreview = Pick<Article, 'title' | 'thumbnail'>;
```

Comment:
Would be good to have a comment there that says that this creates a new type with the chosen properties from `Article`.